Refactoring workflows and encoding profiles

Arnei opened the pull request:
Based on and replaces #1504, as the original author cannot find the time to finish. For more details, see #1504.

To test this, Lars proposed this checklist:

<details>
  <summary>Plan for testing:</summary>
  
* Testing new workflows
 
   * [ ]  Upload presentation and publish directly
   * [ ]  Upload presenter and publish directly
   * [ ]  Upload two streams and publish directly
   * [ ]  Upload presentation, send to editor and publish from there
   * [ ]  Upload presenter, send to editor and publish from there
   * [ ]  Upload two streams, send to editor and publish from there
   * [ ]  Upload presentation, send to editor, cut out element and publish
   * [ ]  Upload presenter, send to editor, cut out element and publish
   * [ ]  Upload two streams, send to editor, cut out element and publish
   * [ ]  Retract event and publish it again
   * [ ]  Republish metadata
   * [ ]  Upload via studio (if the new workflow is used directly, needs more tests otherwise)

* Testing the upgrade path
 
   * [ ]  Upload presentation using old workflow, send to editor and publish from there using the new workflow
   * [ ]  Upload presenter using old workflow, send to editor and publish from there using the new workflow
   * [ ]  Upload two streams using old workflow, send to editor and publish from there using the new workflow
   * [ ]  Upload presentation using old workflow, send to editor, cut out element and publish using the new workflow
   * [ ]  Upload presenter using old workflow, send to editor, cut out element and publish using the new workflow
   * [ ]  Upload two streams using old workflow, send to editor, cut out element and publish using the new workflow
   * [ ]  Retract event published using old workflow and publish it again using the new workflow
   * [ ]  Republish metadata of event published using the old workflow

</details>

Regarding unanswered comments on the previous PR:
> I ran all the listed tests and only Retract event published using old workflow and publish it again using the new workflow does not work. Retracting from engage and publishing to engage again works. But the event is not retracted from the external api and oai-pmh.

The new workflows only publishes to or retracts from engage per default. Users are expected to add other operations to the workflows themselves, based on the publication channels they wish to support. So if you previously have published to the external api and oai-pmh, you should add the related retract operations to your retract workflow.

> Opencast has some encoding profiles hardcoded.

This PR adds any hardcoded profile that I found back to `opencast.properties`.

wsmirnow committed:
Refactoring workflows and encoding profiles

Reduce the complexity defining an clean and small set of features we (as opencast community) want to provide.

The goals of refactoring:

* Reduce the amount of workflows and encoding profiles as much as possible. Remove all workflows and encoding profiles that…
  * aren't used any more
  * are there for historical reasons
  * are optional, very specific or uncommonly used
  * all workflows of disabled features (per default) should be removed, maybe put to the documentation of the feature itself
* Keep the default set of predefined workflows and encoding profiles small
  * Structure and reuse workflow parts
  * Keep it small

To achive the goals we have to make some assumptions:

* Media quality selection for delivery formats could and should be done automaticly
* We have to reduce the amount of predefined publication channels. In most production scenarios the set is static and does not change from episode to another. Here I decided to go with Engage/Search Index. But I leave a comment in the workflows, where you can put your prefered publication channels.
* Keep the editor preview for ever. To use some important editor features we have to keep the editor preview and encode the delivery media from source.

wsmirnow committed:
Create feed, search and player preview images from 3rd second

In case you have trimmed the video with the editor, the first two seconds are fade in. In this period, the extracted images are a bit darker as the schould be.

mtneug committed:
Use engage plugin name in URL to prevent random ID changes

Depending on the plugin registration order the URL to a specific plugin
might change between restarts. Using the static plugin name prevents
this. Note that the name must therefore be unique. A constant URL helps
with caching.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into arne-continues-workflow-refactoring

Arnei committed:
Merge branch 'develop' into arne-continues-workflow-refactoring

Arnei committed:
Make publishing the default for standard workflow

Arnei committed:
Fixed typo

Arnei committed:
Readded missing prepare-ev encoding profiles

Arnei committed:
Readded missing encoding profiles

- Some encoding profiles are hard coded and therefore got removed accidentally
- If the encoding profiles belong to operations that are not used in the standard workflows, the encoding profiles were instead put in the documentation

Arnei committed:
Consolitated duplicate encoding profiles

Arnei committed:
Consolidated comments

Arnei committed:
Moved HLS workflow and encoding profiles to the docs

Arnei committed:
Added comments

lkiesow commented on etc/encoding/opencast.properties:
```diff
@@ -0,0 +1,243 @@
+profile.player-preview.http.output = image
+profile.player-preview.http.suffix = -player.jpg
+profile.player-preview.http.ffmpeg.command = -i #{in.video.path} -frames:v 1 \
+  -filter:v select=eq(n\\,1)+lt(t\\,#{time}),fps=1,reverse,scale=-2:720 \
```
This should suffer from the same problem fixed by https://github.com/lkiesow/opencast/commit/3a8e413c0efbeee737264aa6b021afead9c3bfce

lkiesow commented on etc/encoding/opencast.properties:
```diff
@@ -0,0 +1,243 @@
+profile.player-slides.http.output = image
+profile.player-slides.http.suffix = .#{time}.jpg
+profile.player-slides.http.ffmpeg.command = -ss #{time} -i #{in.video.path} \
+  -r 1 -frames:v 1 -filter:v yadif,scale=160:-1 \
```
We recently discussed in the technical meeting if we do not want to drop `yadif`. I think most were in favor of dropping it. Here is our chance.

lkiesow commented on etc/encoding/opencast.properties:
```diff
@@ -0,0 +1,243 @@
+    -x264opts keyint=25:min-keyint=25:no-scenecut -maxrate 1200k -bufsize 2400k \
+    -c:a aac -b:a 96k -ac 2 \
+    -movflags faststart #{out.dir}/#{out.name}#{out.suffix.720p-quality} \
+  #{if-height-geq-900} \
```
By now, [Resolution Based Encoding](https://docs.opencast.org/r/9.x/admin/#workflowoperationhandlers/encode-woh/#resolution-based-encoding) also supports `if-width-or-height-geq-<width>-<height>` which may be a better choice?

Arnei replied:
So in this particular example, instead of 

`#{if-height-geq-900}`

it would be more desirable to use

`#{if-width-or-height-geq-1600-900}`

to avoid upscaling videos with sufficient height but absurdly small width?

lkiesow commented on etc/encoding/opencast.properties:
```diff
@@ -0,0 +1,243 @@
+profile.adaptive-parallel.http.suffix.1440p-quality = -1440p.mp4
+profile.adaptive-parallel.http.suffix.2160p-quality = -2160p.mp4
+profile.adaptive-parallel.http.ffmpeg.command.if-height-geq-900 = \
+  -filter:v scale=w=1920:h='trunc(1920/dar/2)*2',setsar=1,fps=25 -pix_fmt yuv420p \
```
We previously used `scale=1280:-2` instead of `scale=1280:1280:force_original_aspect_ratio=decrease:force_divisible_by=2` since we were late to the game and `force_divisible_by`  requires a recent FFmpeg version we did not want to require in a minor version bump.

But since this is for the next major version, I would suggest using a scale filter like this:

```
ffmpeg
  -nostdin -nostats
  -i Spring.webm
  -c:v libx264 -crf 23 -maxrate 1200k -bufsize 2400k -profile:v high -level 4.0 -pix_fmt yuv420p
  -x264opts keyint=25:min-keyint=25:no-scenecut
  -movflags +faststart
  -filter:v scale=1280:1280:force_original_aspect_ratio=decrease:force_divisible_by=2,fps=25
  -c:a aac -b:a 64k -ac 1
  -f mp4 out.mp4
```

Documentation can be found at http://ffmpeg.org/ffmpeg-all.html#Options-85

Arnei replied:
While I'm otherwise absolutely in favor of using it, I'm wondering if `force_divisble_by` is still too new an option. If I install ffmpeg on my Fedora 32 via packet manager, the version I'm getting (4.2) does not yet include it.

lkiesow replied:
It's over a year old now, we control most FFmpeg builds on servers and can just require a specific version for a new major version.

EDIT: It's actually even from 2019

lkiesow commented on etc/encoding/opencast.properties:
```diff
@@ -0,0 +1,243 @@
+profile.image-movie.work.output = visual
+profile.image-movie.work.suffix = -image-video.mp4
+profile.image-movie.work.ffmpeg.command = -loop 1 -i #{in.video.path} \
+  -c:v libx264 -r 30 -t #{time} -pix_fmt yuv420p \
```
This may result in:

```
[libx264 @ 0x509bac0] width not divisible by 2 (225x225)                                                                                                                                                                                      
Error initializing output stream 0:0 -- Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
```

Arnei replied:
This encoding profile is also used in the partial import WOH for generating gap fillers before concatenation, I assume it would not be wise to change the resolution of those. Maybe it would be better to create a seperate encoding profile for the partial import WOH.

Edit: Nevermind, just realized this doesn't matter.

lkiesow commented on etc/encoding/opencast.properties:
```diff
@@ -0,0 +1,243 @@
+profile.mp4-preview.dual.http.ffmpeg.command = -i #{in.video.path} \
+  #{compositeCommand} \
+  -c:v libx264 -preset veryfast -crf 23 -profile:v baseline -pix_fmt yuv420p -tune film \
+  -c:a aac -strict -2 -ar 22050 -ab 64k \
```
Why `-strict -2`? That shouldn't be necessary any longer (was years ago when the `aac` encoder was marked experimental).

lkiesow commented on etc/encoding/opencast.properties:
```diff
@@ -0,0 +1,243 @@
+profile.mp4-preview.http.ffmpeg.command = -i #{in.video.path} \
+  -filter:v yadif,scale=-2:360 \
+  -c:v libx264 -preset veryfast -crf 23 -profile:v baseline -pix_fmt yuv420p -tune film \
+  -c:a aac -strict -2 -ar 22050 -ab 64k \
```
```suggestion
  -c:a aac -ar 22050 -ab 64k \
```

lkiesow commented on etc/workflows/publish.xml:
```diff
@@ -181,21 +63,16 @@
+        <!-- On systems with shared workspace or working file repository -->
+        <!-- you want to set this option to false. -->
         <configuration key="delete-external">true</configuration>
-        <!-- FixMe Don't clean up ACLs until workflow service no longer looks for them in the WFR. -->
         <configuration key="preserve-flavors">security/*</configuration>
```
Do we still need this? Removing the comment suggest that the problem is fixed? In that case, we should  do a full cleanup.

Arnei replied:
Seems like the ACLs are still required for parsing ActiveMQ messages, so we have to keep them.

Arnei committed:
Avoid `fps=1`

Explained in lkiesow@3a8e413

Arnei committed:
Drop ffmpeg filter yadif

Arnei committed:
Remove `strict -2` for ffmpeg aac

Not necessary anymore as the aac encoder is no longer experimental

Arnei committed:
Add back comment on why the preserve-flavors key is required in cleanup

Arnei committed:
Avoid uneven width error in image-move.work encoding profile by cropping

Arnei committed:
Updated auto resolution encoding with width checks

Arnei committed:
Changed resolution scaling for standard encoding profiles to use force_divisble_by

Arnei committed:
Added missing newline

gregorydlogan commented:
FTR, this does *not* work with the current Debian packaged ffmpeg.

Arnei committed:
Fixed typo in encoding profiles

Arnei committed:
Merge  'develop' into 'arne-continues-workflow-refactoring'

Arnei committed:
Moved amberscript workflows to the docs

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into arne-continues-workflow-refactoring

Arnei committed:
Reworked Partial Import WOH to use concat demuxer

The concatenation in the Partial Import WOH can require a lot of working memory. Using the concat demuxer instead of concat filter should lower the load considerably.

To that end, all source videos need to be pre-encoded. This commit enforces the use of a preencode encoding profile. It also adds an encoding profile that ensures the necessary prerequisites for concat demuxing.

Arnei committed:
Fixed automated tests

Arnei committed:
Update etc/encoding/opencast-movies.properties

Co-authored-by: Lars Kiesow <lkiesow@uos.de>

lkiesow committed:
Fixed admin interface tests in certain timezones

This patch fixes an admin interface unit test which fails in certain timezones
with active daylight saving time (e.g. CEST).

The error is:

```
       Chrome Headless 93.0.4535.3 (Linux x86_64) adminNg.directives.adminNgTableFilter #selectFilterPeriodValue with both from- and to dates sets the time period filter value (accounting for localized day) FAILED
[INFO]  Expected spy put to have been called with:
[INFO]    [ 'filter', 'furniture', undefined, '2014-12-30T23:00:00.001Z/2015-01-02T22:59:59.999Z' ]
[INFO]  but actual calls were:
[INFO]    [ 'filter', 'furniture', undefined, '2014-12-31T23:00:00.001Z/2015-01-02T22:59:59.999Z' ].
[INFO]
[INFO]  Call 0:
[INFO]    Expected $[3] = '2014-12-31T23:00:00.001Z/2015-01-02T22:59:59.999Z' to equal '2014-12-30T23:00:00.001Z/2015-01-02T22:59:59.999Z'.
[INFO]  Error: Expected spy put to have been called with:
[INFO]    [ 'filter', 'furniture', undefined, '2014-12-30T23:00:00.001Z/2015-01-02T22:59:59.999Z' ]
[INFO]  but actual calls were:
[INFO]    [ 'filter', 'furniture', undefined, '2014-12-31T23:00:00.001Z/2015-01-02T22:59:59.999Z' ].
[INFO]  Call 0:
[INFO]    Expected $[3] = '2014-12-31T23:00:00.001Z/2015-01-02T22:59:59.999Z' to equal '2014-12-30T23:00:00.001Z/2015-01-02T22:59:59.999Z'.
[INFO]      at <Jasmine>
[INFO]      at UserContext.<anonymous> (test/unit/shared/directives/tableFilterDirectiveSpec.js:206:37)
[INFO]      at <Jasmine>
       Firefox 89.0 (Fedora 0.0.0) adminNg.directives.adminNgTableFilter #selectFilterPeriodValue with both from- and to dates sets the time period filter value (accounting for localized day) FAILED
[INFO]  Expected spy put to have been called with:
[INFO]    [ 'filter', 'furniture', undefined, '2014-12-30T23:00:00.001Z/2015-01-02T22:59:59.999Z' ]
[INFO]  but actual calls were:
[INFO]    [ 'filter', 'furniture', undefined, '2014-12-31T23:00:00.001Z/2015-01-02T22:59:59.999Z' ].
[INFO]
[INFO]  Call 0:
[INFO]    Expected $[3] = '2014-12-31T23:00:00.001Z/2015-01-02T22:59:59.999Z' to equal '2014-12-30T23:00:00.001Z/2015-01-02T22:59:59.999Z'.
[INFO]  <Jasmine>
[INFO]  @test/unit/shared/directives/tableFilterDirectiveSpec.js:206:37
[INFO]  <Jasmine>
       Chrome Headless 93.0.4535.3 (Linux x86_64): Executed 739 of 761 (1 FAILED) (skipped 22) (8.758 secs / 7.689 secs)
[INFO] Firefox 89.0 (Fedora 0.0.0): Executed 739 of 761 (1 FAILED) (skipped 22) (11.939 secs / 10.721 secs)
[INFO] TOTAL: 2 FAILED, 1476 SUCCESS
```

npmali committed:
Changed asset mime type  length from 64 to 255

npmali committed:
Changed 9_to_10 to 10_to_11

mtneug committed:
Handle wildcard target flavors in execute-many WOH

The execute-many WOH could not handle wildcard target flavors before.

Arnei committed:
Merge remote-tracking branch 'upstream/r/10.x' into partial-import-encoding-concat

# Conflicts:
#	etc/encoding/opencast-movies.properties

lkiesow committed:
Simplify getUserIdRole

This patch just makes the code in `getUserIdRole)=` a bit simpler. No
functional changes.

mtneug committed:
Remove unused code

mtneug committed:
Use constants instead of hard coded strings

mtneug committed:
Check if S3 tmp directory exists before cleaning it

mtneug committed:
Remove restore job from S3 distribution

mtneug committed:
Fix create new event in admin UI when multiple extended catalogs are used

The admin UI incorrectly assumes that the catalog fields are indexed
continuously which is only the case when only one extended catalog
exists. This searches for the correct field by its ID.

JulianKniephoff committed:
Provide a minimal "No Op" workflow for debugging and maintenance

This can be useful sometimes, for example when parts of our indices
concerning workflows get out of sync with the database, and you see
inconsistent information about the state of a workflow in the events
table vs. the event details. In a case like that, you can just run
some workflow using the `/workflow/start` API to force Opencast
to go through all the proper channels to update the database
and the indices in a (hopefully) consistent manner.

Having a workflow for that purpose handy is thus often useful,
so admins don't have to have it floating around somewhere external,
or come up with the definition from scratch every time. You can just
specify `noop` as a parameter to the `/workflow/start` endpoint.
Also, this workflow intentionally has no `tags` so that it doesn't
show up anywhere in the UI to confuse people.

Arnei committed:
Set jobLoad to 0 when jobCache is empty for very small jobLoads

dependabot[bot] committed:
Bump coffeescript from 2.5.1 to 2.6.0 in /modules/admin-ui-frontend

Bumps [coffeescript](https://github.com/jashkenas/coffeescript) from 2.5.1 to 2.6.0.
- [Release notes](https://github.com/jashkenas/coffeescript/releases)
- [Commits](https://github.com/jashkenas/coffeescript/compare/2.5.1...2.6.0)

---
updated-dependencies:
- dependency-name: coffeescript
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump xmlsec from 2.1.4 to 2.1.7

Bumps xmlsec from 2.1.4 to 2.1.7.

---
updated-dependencies:
- dependency-name: org.apache.santuario:xmlsec
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

wsmirnow committed:
Amberscript attach WOH improvements

The Ambersript attach WOH dous not match documentation. Target flavor and captions format should be optional. Setting target format should not result in multiple captions in the media package.

wsmirnow committed:
Show link to annotation tool in the series LTI tool

The annotate option enable the button to the annotation tool for each episode of the series.

gregorydlogan committed:
Merge pull request #3003 from Arnei/always-set-jobLoad-to-0

Set jobLoad to 0 when jobCache is empty for very small jobLoads

Arnei committed:
Added comment fetching to event fetching

The "/event.json" admin ui endpoint now adds comments to the event. This replaces comment fetching in the frontend, which caused serious load and made the notes column unusable on larger systems.

Supposedly a good solution to this problem would be to add comments to the index, but that is beyond my capabilities atm.

dependabot[bot] committed:
Bump react-bootstrap from 1.6.3 to 1.6.4 in /modules/lti

Bumps [react-bootstrap](https://github.com/react-bootstrap/react-bootstrap) from 1.6.3 to 1.6.4.
- [Release notes](https://github.com/react-bootstrap/react-bootstrap/releases)
- [Changelog](https://github.com/react-bootstrap/react-bootstrap/blob/v1.6.4/CHANGELOG.md)
- [Commits](https://github.com/react-bootstrap/react-bootstrap/compare/v1.6.3...v1.6.4)

---
updated-dependencies:
- dependency-name: react-bootstrap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump i18next from 21.1.1 to 21.2.0 in /modules/lti

Bumps [i18next](https://github.com/i18next/i18next) from 21.1.1 to 21.2.0.
- [Release notes](https://github.com/i18next/i18next/releases)
- [Changelog](https://github.com/i18next/i18next/blob/master/CHANGELOG.md)
- [Commits](https://github.com/i18next/i18next/compare/v21.1.1...v21.2.0)

---
updated-dependencies:
- dependency-name: i18next
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @types/jest from 27.0.1 to 27.0.2 in /modules/lti

Bumps [@types/jest](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jest) from 27.0.1 to 27.0.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jest)

---
updated-dependencies:
- dependency-name: "@types/jest"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Arnei committed:
Merge pull request #2992 from mtneug/fix-ui-multiple-catalogs

Fix create new event in admin UI when multiple extended catalogs are used

Arnei committed:
Merge branch 'develop' into arne-continues-workflow-refactoring

Arnei committed:
Merge branch 'develop' into arne-continues-workflow-refactoring

dependabot[bot] committed:
Bump axios from 0.21.2 to 0.22.0 in /modules/lti

Bumps [axios](https://github.com/axios/axios) from 0.21.2 to 0.22.0.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/master/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v0.21.2...v0.22.0)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Lars Kiesow <lkiesow@uos.de>

lkiesow committed:
Merge r/10.x into develop

dependabot[bot] committed:
Bump @types/react from 17.0.20 to 17.0.26 in /modules/lti

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 17.0.20 to 17.0.26.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

wsmirnow committed:
LTI series tool documentation updated

dependabot[bot] committed:
Bump @types/node from 16.10.1 to 16.10.2 in /modules/lti

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.10.1 to 16.10.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

lkiesow committed:
Deactivate Parallel Builds in CI

This patch deactivates the recently activated parallel builds in the CI
system (GitHub Actions). Speed is not so much a concern there since we
need to wait at least to the next Technical Meeting for approval anyway.
Thus, we do not gain much by activating this.

But parallel Maven builds make identifying reasons for errors
incredibly hard. Especially if they happen in slow building modules
(e.g. in some JavaScript builds). Before you would just look at the end
of the build log. Now you need to scroll through thousands of lines of
logs to identify the root cause for a build failure. That is extremely
annoying as a reviewer.

lkiesow committed:
Point out memory limits

This patch adds a note about memory limits to the 9.x upgrade guide, since it is the first version with external Elasticsearch.
This closes #3016 which could not be merged.

mtneug committed:
Merge branch 'r/9.x' into r/10.x

* r/9.x:
  Point out memory limits
  Fix create new event in admin UI when multiple extended catalogs are used

mtneug committed:
Merge branch 'r/10.x' into develop

* r/10.x:
  Point out memory limits
  Fix create new event in admin UI when multiple extended catalogs are used

Commit:
Automatically update translation keys (develop)

dependabot[bot] committed:
Bump html-validate from 6.0.2 to 6.1.0 in /modules/admin-ui-frontend

Bumps [html-validate](https://gitlab.com/html-validate/html-validate) from 6.0.2 to 6.1.0.
- [Release notes](https://gitlab.com/html-validate/html-validate/tags)
- [Changelog](https://gitlab.com/html-validate/html-validate/blob/master/CHANGELOG.md)
- [Commits](https://gitlab.com/html-validate/html-validate/compare/v6.0.2...v6.1.0)

---
updated-dependencies:
- dependency-name: html-validate
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump grunt-contrib-jshint in /modules/admin-ui-frontend

Bumps [grunt-contrib-jshint](https://github.com/gruntjs/grunt-contrib-jshint) from 3.0.0 to 3.1.1.
- [Release notes](https://github.com/gruntjs/grunt-contrib-jshint/releases)
- [Changelog](https://github.com/gruntjs/grunt-contrib-jshint/blob/main/CHANGELOG)
- [Commits](https://github.com/gruntjs/grunt-contrib-jshint/commits/v3.1.1)

---
updated-dependencies:
- dependency-name: grunt-contrib-jshint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mtneug committed:
Set external API version v1.7.0 as default (#3005)

API versioning is hard.

I forgot to set v1.7.0 as default version and to include it in (#2814)  the
versions request.

Arnei committed:
Fixed visual comment duplication glitch

gregorydlogan committed:
Merge branch 'amberscript-woh-improvements' of wsmirnow/opencast into r/10.x

Pull request #3013
  Amberscript attach WOH improvements

gregorydlogan committed:
Merge branch 'oc_assets_mimetype_length_to_varchat255_2215' of npmali/opencast into develop

Pull request #2744
  Changed asset mime type  length from 64 to 255

gregorydlogan committed:
Removing SQL upgrade patch from old versions.  If you've got an OC 8 or earlier version you *must* go through 9 and 10 first since those versions have a number of other DB patches as well.

Arnei committed:
Merge pull request #2956 from wsmirnow/lti-series-tool-annotate-button

Show link to annotation tool in the series LTI tool

dependabot[bot] committed:
Bump markdownlint-cli from 0.28.1 to 0.29.0 in /docs/guides

Bumps [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli) from 0.28.1 to 0.29.0.
- [Release notes](https://github.com/igorshubovych/markdownlint-cli/releases)
- [Commits](https://github.com/igorshubovych/markdownlint-cli/compare/v0.28.1...v0.29.0)

---
updated-dependencies:
- dependency-name: markdownlint-cli
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump coffeescript from 2.6.0 to 2.6.1 in /modules/admin-ui-frontend

Bumps [coffeescript](https://github.com/jashkenas/coffeescript) from 2.6.0 to 2.6.1.
- [Release notes](https://github.com/jashkenas/coffeescript/releases)
- [Commits](https://github.com/jashkenas/coffeescript/compare/2.6.0...2.6.1)

---
updated-dependencies:
- dependency-name: coffeescript
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

gregorydlogan committed:
Merge pull request #3040 from lkiesow/no-parallel

Deactivate Parallel Builds in CI

Arnei committed:
Only command the backend to fetch the comments for the events displayed in the Admin UI IF the notes columns is actually shown

Arnei committed:
Fix test errors

dependabot[bot] committed:
Bump grunt-contrib-concat in /modules/admin-ui-frontend

Bumps [grunt-contrib-concat](https://github.com/gruntjs/grunt-contrib-concat) from 1.0.1 to 2.0.0.
- [Release notes](https://github.com/gruntjs/grunt-contrib-concat/releases)
- [Changelog](https://github.com/gruntjs/grunt-contrib-concat/blob/main/CHANGELOG)
- [Commits](https://github.com/gruntjs/grunt-contrib-concat/compare/v1.0.1...v2.0.0)

---
updated-dependencies:
- dependency-name: grunt-contrib-concat
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Automatically update translation keys (develop)

Commit:
Automatically update translation keys (r/10.x)

Commit:
Automatically update translation keys (r/9.x)

dependabot[bot] committed:
Bump eslint from 7.32.0 to 8.0.0 in /modules/runtime-info-ui

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.0.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.0.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

karendolan committed:
Opencast 10x docs show old "compose" instead of new "encode" name #3059 (ref #1421)

mtneug committed:
Remove unused constants

Remove constants that are not used in execute-once WOH

dependabot[bot] committed:
Bump eslint from 7.32.0 to 8.0.0 in /modules/runtime-info-ui-ng

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.0.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.0.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 7.32.0 to 8.0.0 in /modules/admin-ui-frontend

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.0.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.0.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

JulianKniephoff committed:
Updated Release Notes

lkiesow committed:
Merge r/9.x into r/10.x

lkiesow committed:
Merge r/10.x into develop

lkiesow committed:
Deactivate security-jwt by default

As discussed at the technical meeting, due to potential problems causing
Opencast to not start up properly, this patch disables the module
`security-jwt` by default, allowing users to activate them using the
`opencast-security-jwt` Karaf feature.

We hope to have this enabled by default again in Opencast 11.

lkiesow committed:
Fix LTI Tool Build

In TypeScript 4.4.3 the type declarations for `top` has changed to
`WindowProxy | null`. That is why pull request #2956 broke the build
once it was merged into `develop`. `develop` already uses the new
TypeScript version. The result is:

```
Failed to compile.

/home/lars/dev/opencast/modules/lti/src/components/Series.tsx
TypeScript error in /home/lars/dev/opencast/modules/lti/src/components/Series.tsx(184,9):
Object is possibly 'null'.  TS2531

    182 |
    183 |     annotateEpisodeCallback(id: string) {
  > 184 |         window.top.location.href = "/annotation-tool/index.html?id=" + id
        |         ^
    185 |     }
    186 |
    187 |     downloadEventCallback(track: Track) {

```

According to the specification, `window.top` may never be `null` in a
bowser context. That is why it should be safe to just tell TypeScript to
ignore that this type could be `null`.

This fixes the build.

gregorydlogan committed:
Merge pull request #3066 from lkiesow/fix-lti-build

Fix LTI Tool Build

dependabot[bot] committed:
Bump eslint from 7.32.0 to 8.0.0 in /modules/engage-theodul-core

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.0.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.0.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 7.32.0 to 8.0.0 in /modules/engage-ui

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.0.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.0.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 7.32.0 to 8.0.0 in /modules/engage-paella-player

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.0.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.0.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mtneug committed:
Merge pull request #3065 from lkiesow/deactivate-secuirity-jwt-by-default

Deactivate security-jwt by default

gregorydlogan committed:
Merge branch 'partial-import-encoding-concat' of Arnei/opencast into r/10.x

Pull request #2669
  Reworked Partial Import WOH to use concat demuxer

gregorydlogan committed:
Merge branch 'noop-workflow' of JulianKniephoff/opencast into r/10.x

Pull request #2994
  Provide a minimal "No Op" workflow for debugging and maintenance

gregorydlogan committed:
Merge branch 'notes-fetch-comments-in-backend' of Arnei/opencast into r/10.x

Pull request #3044
  Notes fetch comments in backend

gregorydlogan committed:
Updating changelog and release notes

dependabot[bot] committed:
Bump eslint from 8.0.0 to 8.0.1 in /modules/engage-paella-player

Bumps [eslint](https://github.com/eslint/eslint) from 8.0.0 to 8.0.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.0.0...v8.0.1)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 8.0.0 to 8.0.1 in /modules/admin-ui-frontend

Bumps [eslint](https://github.com/eslint/eslint) from 8.0.0 to 8.0.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.0.0...v8.0.1)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jasmine-core from 3.9.0 to 3.10.0 in /modules/admin-ui-frontend

Bumps [jasmine-core](https://github.com/jasmine/jasmine) from 3.9.0 to 3.10.0.
- [Release notes](https://github.com/jasmine/jasmine/releases)
- [Changelog](https://github.com/jasmine/jasmine/blob/main/RELEASE.md)
- [Commits](https://github.com/jasmine/jasmine/compare/v3.9.0...v3.10.0)

---
updated-dependencies:
- dependency-name: jasmine-core
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 8.0.0 to 8.0.1 in /modules/engage-ui

Bumps [eslint](https://github.com/eslint/eslint) from 8.0.0 to 8.0.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.0.0...v8.0.1)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 8.0.0 to 8.0.1 in /modules/engage-theodul-core

Bumps [eslint](https://github.com/eslint/eslint) from 8.0.0 to 8.0.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.0.0...v8.0.1)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 8.0.0 to 8.0.1 in /modules/runtime-info-ui-ng

Bumps [eslint](https://github.com/eslint/eslint) from 8.0.0 to 8.0.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.0.0...v8.0.1)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 8.0.0 to 8.0.1 in /modules/runtime-info-ui

Bumps [eslint](https://github.com/eslint/eslint) from 8.0.0 to 8.0.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.0.0...v8.0.1)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

KatrinIhler committed:
Merge pull request #3067 from opencast/dependabot/npm_and_yarn/modules/engage-paella-player/eslint-8.0.1

Bump eslint from 8.0.0 to 8.0.1 in /modules/engage-paella-player

KatrinIhler committed:
Merge pull request #3068 from opencast/dependabot/npm_and_yarn/modules/admin-ui-frontend/eslint-8.0.1

Bump eslint from 8.0.0 to 8.0.1 in /modules/admin-ui-frontend

KatrinIhler committed:
Merge pull request #3073 from opencast/dependabot/npm_and_yarn/modules/runtime-info-ui/eslint-8.0.1

Bump eslint from 8.0.0 to 8.0.1 in /modules/runtime-info-ui

KatrinIhler committed:
Merge pull request #3072 from opencast/dependabot/npm_and_yarn/modules/runtime-info-ui-ng/eslint-8.0.1

Bump eslint from 8.0.0 to 8.0.1 in /modules/runtime-info-ui-ng

KatrinIhler committed:
Merge pull request #3071 from opencast/dependabot/npm_and_yarn/modules/engage-theodul-core/eslint-8.0.1

Bump eslint from 8.0.0 to 8.0.1 in /modules/engage-theodul-core

KatrinIhler committed:
Merge pull request #3070 from opencast/dependabot/npm_and_yarn/modules/engage-ui/eslint-8.0.1

Bump eslint from 8.0.0 to 8.0.1 in /modules/engage-ui

dependabot[bot] committed:
Bump autoprefixer from 9.8.7 to 9.8.8 in /modules/admin-ui-frontend

Bumps [autoprefixer](https://github.com/postcss/autoprefixer) from 9.8.7 to 9.8.8.
- [Release notes](https://github.com/postcss/autoprefixer/releases)
- [Changelog](https://github.com/postcss/autoprefixer/blob/9.8.8/CHANGELOG.md)
- [Commits](https://github.com/postcss/autoprefixer/compare/9.8.7...9.8.8)

---
updated-dependencies:
- dependency-name: autoprefixer
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

KatrinIhler committed:
Merge pull request #3063 from mtneug/exec-wildcard

Handle wildcard target flavors in execute-many WOH

KatrinIhler committed:
Sign publication URL of events in External API (#3002)

The External API already offers a parameter to pre-sign distribution
URLs when requesting one or more events with their publications.
Starting with version 1.7.0 this change will also pre-sign the
publication URLs itselves when that flag is set.

KatrinIhler committed:
Merge pull request #2924 from lkiesow/simplify-userdirectory

Simplify getUserIdRole

lkiesow committed:
Test only with Firefox or Chrome (#2925)

As discussed at today's technical meeting this patch closes #2527 and
replaces it with a patch limiting the tests to Firefox or Chrome so we
know that we actually run in a somewhat known and safe environment.

lkiesow committed:
Test only with Firefox or Chrome (#2925)

As discussed at today's technical meeting this patch closes #2527 and
replaces it with a patch limiting the tests to Firefox or Chrome so we
know that we actually run in a somewhat known and safe environment.

KatrinIhler committed:
Merge pull request #2936 from mtneug/cleanup-s3-code

Cleanup S3 code

lkiesow committed:
Merge r/10.x into develop

Arnei committed:
Fixed limit value returned by search service endpoint (#3050)

Previously, the limit value in the response xml/json was guessed at. This changes the value to actual limit that was being used.

Arnei committed:
Removed ManagedService from Admin UI backend (#3049)

The ManagedService interface would cause service restarts on
configuration updates. This PR removes said interface and replaces it
with OSGi annotations.

gregorydlogan committed:
Remove staticweave plugin (#3031)

Removing staticweave code.  This plugin is not maintained and also not threadsafe.  Nor does it appear to be actually enabled correctly.

lkiesow committed:
Merge r/9.x into r/10.x

lkiesow committed:
Merge r/10.x into develop

lkiesow committed:
Merge with develop

lkiesow commented:
Fixed merge conflicts by merging with `develop`.

gregorydlogan commented:
This may or may not cause issues with #3084, depending on what's actually in these changes (I haven't looked).

lkiesow committed:
Minor Workflow Changes

This patch contains a few minor modifications to the new workflows.
The changes are mainly:

- Documentation/Upgrade guide
- Workflow descriptions
- Removed or collapsed some tag operations
- Removed some unnecessary operations from fast

There is still room for optimizations. For example, you could remove
some more tag operations and there is an exceeding amount of analyze
tracks, but that's maybe something for later.

Additionally, we may want to remove the timelinepreview and cover-image
operations. But that needs additional discussions.
